octobot_trading/personal_data/portfolios/types/future_portfolio.py

```diff
@@ -43,26 +43,37 @@ def update_portfolio_data_from_order(self, order):
 
             # When inverse contract, decrease a currency market equivalent quantity from currency balance
             if pair_future_contract.is_inverse_contract():
-                total_update_quantity = real_order_quantity / order.filled_price
                 fees_update_quantity = -order.get_total_fees(order.currency)
+                total_update_quantity = real_order_quantity / order.filled_price
+                position_margin_update = real_order_quantity / position_instance.entry_price - fees_update_quantity
                 self._update_future_portfolio_data(
                     order.currency,
                     wallet_value=fees_update_quantity,
                     order_margin_value=-total_update_quantity if has_increased_position_size else constants.ZERO,
-                    position_margin_value=total_update_quantity
-                    if has_increased_position_size else -total_update_quantity)
+                    position_margin_value=position_margin_update
+                    if has_increased_position_size else -position_margin_update)
 
             # When non-inverse contract, decrease directly market quantity
             else:
                 # decrease market quantity from market available balance
+                fees_update_quantity = -(order.get_total_fees(order.market) +
+                                         order.get_total_fees(order.currency) * order.filled_price)
```

Comment:
I don't understand this line, how fees could only depend on `filled_price` ?

Reply:
When a fee is in currency, we need to convert it in ref market otherwise it's ignored in the current system (in BTC/USDT, only USDT asset is updated, therefore BTC fees are not taken into account). (this part is still a wip)